frontend/src/pages/admin/index.svelte

```diff
@@ -486,6 +810,410 @@
 				{/if}
 			</div>
 		</div>
+		<div
+			class="mx-2 my-4 col-span-12 grid grid-cols-12 gap-4 {activeView ==
+			2
+				? 'block'
+				: 'hidden'}"
+		>
+			{#key activeView}
+				<div class="col-span-12 flex flex-wrap justify-between gap-y-2">
+					<h1 class="text-3xl font-bold">Reports & Statistics</h1>
+					<div class="flex gap-x-4 items-end flex-wrap">
+						<div>
+							<label class="label" for="">
+								<span class="label-text">Date range</span>
+								<span class="label-text pr-0.5">Active</span>
+							</label>
+							<div
+								class="flex items-center input input-bordered divide-x divide-gray-300 dark:divide-gray-500"
+							>
+								<div class="dropdown dropdown-end">
+									<label
+										class="pr-2 flex items-center cursor-pointer"
+										for=""
+										tabindex="0"
+									>
+										<span class="text-sm"
+											>{dateRangeText}</span
+										>
+										<span
+											class="icon-calendar-range text-xl ml-3"
+										/>
+									</label>
+									<ul
+										tabindex="0"
+										class="dropdown-content menu p-2 mt-4 shadow bg-base-100 rounded-box w-52"
+									>
+										<Input
+											type="date"
+											label="From"
+											bind:value={dateRangeValues.from}
+											error={dateRangeIsValid()
+												? dateRangeValues.from == ''
+													? false
+													: null
+												: `"From" can't be after "to"`}
+											max={TODAY}
+										/>
+										<Input
+											type="date"
+											label="To"
+											bind:value={dateRangeValues.to}
+											error={dateRangeIsValid()
+												? dateRangeValues.to == ''
+													? false
+													: null
+												: `"To" can't be before "from"`}
+											max={TODAY}
+										/>
+										<button
+											class="btn btn-sm btn-primary mt-2"
+											on:click={applyRange}
+											disabled={!dateRangeIsValid()}
+											>Apply date range</button
+										>
+									</ul>
+								</div>
+								<div class="pl-2 h-full flex items-center">
+									<input
+										type="checkbox"
+										class="checkbox"
+										bind:checked={dateRangeActive}
+										on:change={updateReports}
+									/>
+								</div>
+							</div>
+						</div>
+						<div>
+							<Input
```

Comment:
Why is there no option for all? Also, appears the # of entries shown has no server side validation. Is this intentional?

Reply:
There isn't an option for all because displaying all entries in a single page could get very big and the backend would also need changes to support that use (there is no way to do that currently). However, you can view all entries by downloading the CSV for the thing you're looking at. This could probably be made more clear, but I don't know how.
Would you be fine with adding bigger presets? That can just be done on the frontend easily.

I'm not totally clear what you mean by "no server side validation", all endpoints that can be filtered by count do validate the query parameter to make sure it's a valid number. If you mean we don't enforce the specific values given (they don't have to be 5, 10, 20, or 50) this is intentional and I don't see why we would need to restrict use to those values only.

Reply:
Ah ok, I didn't know the CSV included everything. I think adding an 100 preset might be a good idea but if the CSV includes everything isn't a big deal. I wasn't sure if we wanted to make it so the count can't be too big to prevent lag or something, but if you don't think it's a concern then it's fine with me.